main_window/rocket_data.py

```diff
@@ -100,8 +108,11 @@ def add_bundle(self, full_address: FullAddress, incoming_data: Dict[DataEntryIds
             # write the data
             for data_id in incoming_data:
                 key = DataEntryKey(full_address, data_id)
-                self.existing_entry_keys.add(key)
+
+                if key not in set(self.keyset.keys()):
```

Comment:
```suggestion
                if key not in self.keyset:
```
I know this works with a normal dict, idk about ordereddict though.

Reply:
Oh yes, sorry for this one .key's is unnecessary.